src/NuGet.Core/NuGet.Build.Tasks/RestoreTask.cs

```diff
@@ -88,6 +88,14 @@ public class RestoreTask : Microsoft.Build.Utilities.Task, ICancelableTask, IDis
         [Output]
         public ITaskItem[] EmbedInBinlog { get; set; }
 
+        /// <summary>
+        /// Gets or sets a value indicating whether to embed files produced by restore in the MSBuild binary logger.
+        /// 0 = Nothing
+        /// 1 = Assets file, g.props, and g.targets
+        /// 2 = dgspec, assets file, g.props, and g.targets
+        /// </summary>
+        public string EmbedFilesInBinlog { get; set; }
```

Comment:
did you consider an enum?

Reply:
I could parse the value as an enum but I wanted to just support true/false plus one more value.  So I settled on:

0/false = off
1/true = default
2 = everything

I could still parse the value into an enum later instead of an integer if you prefer. But I didn't want users to have to specify values like `None`, `Default`, `All`.

Reply:
GIven that this is a debug thing, we can always change it later. I'm not worried about it.